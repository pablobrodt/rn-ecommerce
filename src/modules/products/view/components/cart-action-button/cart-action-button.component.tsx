import { Button } from '@common/view/components/button/button.component';
import { Colors } from '@common/view/constants/colors';
import type { PropsWithStyle } from '@common/view/types/props-with-style';

import { styles } from './cart-action-button.style';
import { CartArrowDown, CartArrowUp } from '@assets/icons';
import { PropsWithTestId } from '@common/view/types/props-with-test-id';

type CartActionButtonProps = {
  isInCart: boolean;
  onPress: () => void;
} & PropsWithStyle &
  PropsWithTestId;

export function CartActionButton({
  style,
  isInCart,
  testId,
  onPress,
}: CartActionButtonProps) {
  const Icon = isInCart ? CartArrowUp : CartArrowDown;
  const backgroundColor = isInCart
    ? Colors.DANGER_REGULAR
    : Colors.SUCCESS_REGULAR;

  function handleOnPress() {
    onPress();
  }

  return (
    <Button
      style={[styles.container, style, { backgroundColor: backgroundColor }]}
      onPress={handleOnPress}
      testId={testId}
    >
      <Icon style={styles.icon} color={Colors.WHITE} />
    </Button>
  );
}
