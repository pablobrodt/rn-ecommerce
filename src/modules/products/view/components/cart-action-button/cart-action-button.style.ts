import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    maxHeight: 44,
  },
  icon: {
    width: 24,
    height: 24,
  },
});
