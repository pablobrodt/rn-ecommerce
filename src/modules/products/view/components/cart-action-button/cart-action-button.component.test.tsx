import { render } from '@testing-library/react-native';

import { CartActionButton } from './cart-action-button.component';

jest.mock('@assets/icons', () => ({
  CartArrowDown: 'CartArrowDown',
  CartArrowUp: 'CartArrowUp',
}));

describe('@Products Cart Action Button Component Tests', () => {
  it('should render CartArrowDown icon and Colors.SUCCESS_REGULAR when not in cart', () => {
    const { toJSON } = render(
      <CartActionButton isInCart={false} onPress={jest.fn()} />,
    );

    const component = toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should render CartArrowUp icon and Colors.DANGER_REGULAR when in cart', () => {
    const { toJSON } = render(
      <CartActionButton isInCart={true} onPress={jest.fn()} />,
    );

    const component = toJSON();

    expect(component).toMatchSnapshot();
  });
});
