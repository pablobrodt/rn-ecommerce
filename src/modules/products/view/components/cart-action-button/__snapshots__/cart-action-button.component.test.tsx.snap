// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Products Cart Action Button Component Tests should render CartArrowDown icon and Colors.SUCCESS_REGULAR when not in cart 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#40a250",
      "borderRadius": 4,
      "flexGrow": 1,
      "justifyContent": "center",
      "maxHeight": 44,
      "minHeight": 44,
      "minWidth": 44,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <CartArrowDown
    color="#FFFFFF"
    style={
      {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;

exports[`@Products Cart Action Button Component Tests should render CartArrowUp icon and Colors.DANGER_REGULAR when in cart 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#c04547",
      "borderRadius": 4,
      "flexGrow": 1,
      "justifyContent": "center",
      "maxHeight": 44,
      "minHeight": 44,
      "minWidth": 44,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <CartArrowUp
    color="#FFFFFF"
    style={
      {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
